components/service/firefox-accounts/src/test/java/mozilla/components/service/fxa/store/FxaAccountManagerIntegrationTest.kt

```diff
@@ -0,0 +1,154 @@
+package mozilla.components.service.fxa.store
+
+import androidx.lifecycle.LifecycleOwner
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.test.runTest
+import mozilla.components.concept.sync.Avatar
+import mozilla.components.concept.sync.ConstellationState
+import mozilla.components.concept.sync.DeviceConstellation
+import mozilla.components.concept.sync.OAuthAccount
+import mozilla.components.concept.sync.Profile
+import mozilla.components.service.fxa.manager.FxaAccountManager
+import mozilla.components.support.test.any
+import mozilla.components.support.test.coMock
+import mozilla.components.support.test.eq
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.whenever
+import org.junit.Before
+import org.junit.Test
+import org.mockito.Mockito.never
+import org.mockito.Mockito.verify
+import java.lang.Exception
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class FxaAccountManagerIntegrationTest {
+
+    private val store = mock<SyncStore>()
```

Comment:
I don't think tests like these are valuable. We are just testing the mocks and not the interaction.

Using a real store is also important because that's what want to test - the real thing.

What our tests should look like is:
1. Invoke or fire an account observer.
2. Add `store.waitUntilIdle()`
3. Assert the `store.state` value is what you expected it to be.
4. (Optionally) add the `CaptureActionsMiddleware` to our test instance of the store if you really want to verify the actions are triggered as well.

You might find some this test class to be a helpful example: https://searchfox.org/mozilla-mobile/source/android-components/components/feature/search/src/test/java/mozilla/components/feature/search/SearchFeatureTest.kt

Reply:
I started a comment about my test philosophy in regards to unit vs integration tests, and halfway through it realized that the name of the code under test is literally "integration" 😂 

Given that, I can definitely see where you're coming from. It does still feel a bit like we'd be missing a bit of tests - unit tests around the reducer itself. Those are so straightforward and reverse-implementation-y that I hadn't included them originally, but I think there's an argument to be made that if I had, then the combined set of unit tests would have effectively captured the behavior suggested by your comment. 

Original start of my reply: 
I kind of fundamentally disagree that this type of test isn't valuable. IMO there's still quite a bit of value in unit tests vs integration tests, especially in that
- they're a form of documentation
- they prevent regressions

Reply:
This is a good discussion to continue in the future though. 🙂